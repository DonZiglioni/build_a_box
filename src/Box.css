.box {
    display: flex;
    flex-direction: row;
    align-items: start;
    padding: 10px;
}

.del-button {
    display: flex;
    align-items: center;
    justify-content: center;
    border-radius: 50%;
    border: none;
    height: 1.5em;
    width: 1.5em;
    margin: .5em;
}