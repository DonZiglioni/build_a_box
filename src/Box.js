import React from "react";
import './Box.css'

//  ** Individual Box Component  **
const Box = ({ id, color, height, width, removeBox }) => {
    return (
        <div className="box">
            <div
                id={id}
                style={{
                    backgroundColor: color,
                    height: height,
                    width: width,
                    display: "flex",
                    justifyContent: "center",
                    alignItems: "center",
                }}>
            </div>
            <button onClick={() => removeBox(id)} className="del-button">x</button>
        </div>
    )
}

export default Box;