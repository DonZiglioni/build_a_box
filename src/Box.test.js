import React from "react";
import { render } from "@testing-library/react"
import Box from "./Box";

it('Renders Successfully', () => {
    render(<Box />);
})

it('Matches Snapshot', () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
})