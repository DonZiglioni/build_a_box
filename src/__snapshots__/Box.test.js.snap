// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<DocumentFragment>
  <div
    class="box"
  >
    <div
      style="display: flex; justify-content: center; align-items: center;"
    />
    <button
      class="del-button"
    >
      x
    </button>
  </div>
</DocumentFragment>
`;
